File: Algorithms/137-Single-Number-II.swift
class Solution {
	func singleNumber(_ nums: [Int]) -> Int {
		var a = 0, b = 0
		
		nums.forEach { num in
			b = (b ^ num) & ~a
			a = (a ^ num) & ~b
		}
		
		return a | b
	}
}

File: Algorithms/137-Single-Number-II.cpp
class Solution {
public:
	int singleNumber(vector<int>& nums) {
		int a = 0, b = 0;
		
		for (int num : nums) {
			b = (b ^ num) & ~a;
			a = (a ^ num) & ~b;
		}
		
		return a | b;
	}
};